File: 60-Permutation Sequence.py
class Solution(object):
    def getPermutation(self, n, k):
        """
        :type n: int
        :type k: int
        :rtype: str
        """           
        
        def factorialFunc(num):
            res = 1
            for i in range(num,0,-1):
                res = res*i
            return res
        
        factorial = factorialFunc(n)
        
        arr = [str(i) for i in range(1, n+1)]
        k-=1
        res = []
        
        while k > 0:
            factorial //= n
            count, k = divmod(k, factorial)
            for i in range(len(arr)):
                if not arr[i]:
                    continue
                if count == 0:
                    res.append(arr[i])
                    arr[i] = None
                    break
                count -= 1
            n-=1
            
        for xx in arr:
            if xx:
                res.append(xx)
        
        return "".join(res)

File: JavaScript/60-Permutation Sequence.js
/**
 * @param {number} n
 * @param {number} k
 * @return {string}
 */
var getPermutation = function(n, k) {
    
    let factorialFunc = (num) =>{
        let res = 1;
        for (let x = 2; x<=num; x++){
            res *= x;
        }
        return res;
    }
    
    let arr = Array(n).fill().map((_, i) => (i+1).toString() );
    let factorial = factorialFunc(n);
    let res = new Array();
    --k;
    
    while (k > 0){
        factorial = factorial/n;
        let index = ~~(k/factorial);
        k = k%factorial;
        for (let i=0; i<arr.length; i++){
            if (arr[i] === null) continue;
            if (index === 0){
                res.push(arr[i]);
                arr[i] = null;
                break;
            }else{
                --index;
            }
        }
        n--;
    }
    
    for (let xx of arr){
        if (xx){
            res.push(xx);
        }
    }
    return res.join("");
};